### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by zoebarth**
Add authentication functionality. Functionality is found in app.js and can be passed down to other pages as needed. Proxy middleware redirects to appropriate local host.

To do in the future:
- Make sure log in and log out redirects to appropriate page

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -8,14 +8,20 @@ import Navbar from './navbar/Navbar.js';
 import testImg from './attraction/Attraction.png';
 
 function App() {
-  const [loggedIn, setLoggedIn] = useState(false);
+  const [authState, setAuthState] = useState({ ready: false });
+
+  fetch('/api/v1/auth')
+    .then((response) => response.json())
+    .then(({ loggedIn, loginUrl, logoutUrl }) =>
+      setAuthState({ ...authState, ready: true, loggedIn, loginUrl, logoutUrl })
```
Comment (Cindyzhang977) [inline frontend/src/App.js:16]: What does ... do?

File: frontend/src/App.js [hunk 2]
```diff
@@ -8,14 +8,20 @@ import Navbar from './navbar/Navbar.js';
...
+    );
```
[the remark this answers is not in the document]
Comment (mecolmg) [reply frontend/src/App.js:17]: Just to explain why this is happening: Without using `useEffect` this fetch is being called on each rerender, and each time it's called you update the state causing an additional rerender, over and over and over again.

File: frontend/src/App.js (hunk 1 above)
Comment (zoebarth) [reply frontend/src/App.js:16]: It's the JS spread syntax https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

File: backend/src/main/java/com/google/sps/servlets/OptimizeServlet.java [hunk 3]
```diff
@@ -28,21 +28,21 @@
 @WebServlet("/optimize")
```
[the remark this answers is not in the document]
Comment (zoebarth) [reply backend/src/main/java/com/google/sps/servlets/OptimizeServlet.java]: I changed it to "/api/v1/optimize" but I'm not sure where to change it in Map.js

**Pull request merged**